### DISCUSSION TITLE
Help with using ROS for RL training

### DISCUSSION THREAD

**Discussion by sesem738** [isaac-sim/IsaacLab#188] [Q&A] [answered]
I am a beginner in Isaac Sim and Orbit. I have a robot with ROS packages for low-level control. I want to train it to perform some tasks with RL. How can I do this? What are some good examples to follow? Should I train with or without ROS integration? How can I find the sim-to-real code for the AnyMal D in the research paper?

**Comment by Mayankm96** [comment, marked as answer]
Hi @sesem738 ,

ROS communication leads to significant overhead, which makes it non-ideal for performant RL training. I advice you to take the following steps:

1. Convert your asset into USD format: https://isaac-orbit.github.io/orbit/source/how-to/import_new_asset.html
2. Create a Orbit configuration class for your robot: https://isaac-orbit.github.io/orbit/source/how-to/write_articulation_cfg.html
3. Check that your robot has been imported correctly and you can control the joints:  https://isaac-orbit.github.io/orbit/source/tutorials/01_assets/run_articulation.html
4. Follow our tutorials on setting up an environment for RL training: https://isaac-orbit.github.io/orbit/source/tutorials/03_envs/index.html

Hope this helps!
